src/complex/Filter/IFilter.hpp

```diff
@@ -35,12 +35,14 @@ class COMPLEX_EXPORT IFilter
     {
       Info = 0,
       Debug,
+      Progress,
       Warning,
       Error
     };
 
     Type type = Type::Info;
     std::string message;
+    int32 progress;
```

Comment:
Do we need a dedicated progress message type? And if so do we need to include the progress value for every message type?